tests/test_browser.py

```diff
@@ -3361,6 +3366,30 @@ def test_modularize_network_error(self):
     os.remove('a.out.wasm')
     self.run_browser('a.html', '...', '/report_result?abort(both async and sync fetching of the wasm failed)')
 
+  def test_modularize_init_error(self):
+    src = open(path_from_root('tests', 'browser_test_init_crash.cpp')).read()
+    create_test_file('test.cpp', src)
+    browser_reporting_js_path = path_from_root('tests', 'browser_reporting.js')
+    self.compile_btest(['test.cpp', '-s', 'MODULARIZE=1', '-s', 'EXPORT_NAME="createModule"', '--extern-pre-js', browser_reporting_js_path])
+    create_test_file('a.html', '''
+      <script src="a.out.js"></script>
+      <script>
+        if (typeof window === 'object' && window) {
+          window.addEventListener('unhandledrejection', function(event) {
+            reportResultToServer("Unhandled promise rejection: " + event.reason.message);
+          });
+        }
+        createModule()
+          .then(() => {
+            reportResultToServer("Module creation succeeded when it should have failed");
+          })
+          .catch(err => {
+            reportResultToServer(err.message.slice(0, 88));
+          });
+      </script>
+    ''')
+    self.run_browser('a.html', '...', '/report_result?abort(Runtime error: The application has corrupted its heap memory area (address zero)!)')
```

Comment:
This error text may change in the future. How about in the C++ throwing a JS exception with specific fixed text? That is,
```cpp
#include <emscripten.h>
...
EM_ASM({
  throw "intentional error to test rejection";
});
```

Reply:
Great idea! 😊